packages/by-name/kata/genpolicy/package.nix

```diff
@@ -114,6 +114,41 @@ rustPlatform.buildRustPackage rec {
         runHook postInstall
       '';
     };
+
+    # Builds a policy that responds to all known agent requests with a fixed response.
+    # Such a policy is useful for propagating errors from the guest to the runtime.
+    # The actual message needs to be appended to the rules, similar to policy_data in genpolicy.
+    rules-deny-with-message = stdenvNoCC.mkDerivation {
+      name = "${pname}-${version}-rules-deny-with-message";
+      inherit src;
+
+      phases = [
+        "unpackPhase"
+        "patchPhase"
+        "buildPhase"
+        "installPhase"
+      ];
+      buildPhase = ''
+          awk '
+            # header
+            BEGIN { print "package agent_policy\n" }
+
+            # default deny rule for the request
+            /^message.*Request/ { print "default",$2,":= false" }
+            # denying rule that calls print_message
+            /^message.*Request/ { print $2,"{ print_message }" }
+
+            # implementation of print_message (`message := "foo"` needs to be appended by the user)
+            END { print "\nprint_message {\n  print(\"Internal error:\", message)\n  false\n}" }
+            ' \
+            $src/src/libs/protocols/protos/agent.proto >deny-with-message.rego
+      '';
+      installPhase = ''
+        runHook preInstall
+        install -D deny-with-message.rego $out/genpolicy-rules.rego
+        runHook postInstall
+      '';
```

Comment:
Why is this part of the genpolicy package?

And what is the relationship to deny-with-message.rego in initdata-processor/policy/assets? Looks like this is duplicated.

Reply:
I thought it would make sense to put it next to the existing policy-generating derivations. Maybe it doesn't? `deny-with-message.rego` is supposed to overwrite `initdata-processor/policy/assets/deny-with-message.rego` (which I thought needs to be checked in anyway, like [genpolicy settings](https://github.com/edgelesssys/contrast/blob/5497a7d6f1a902cced5ce87bf31e3d62e6cedebb/cli/genpolicy/assets/genpolicy-settings-kata.json)). Should I put it into the initdata-processor Nix package?